File: Python/1025 - Onde está o Mármore.py
caso = 0
while True:
    entrada = input().strip()
    if not entrada:
        continue

    N, Q = map(int, entrada.split())

    if N == 0 and Q == 0:
        break

    marmores = []
    for _ in range(N):
        marmores.append(int(input().strip()))
    marmores.sort()

    caso += 1
    print(f"CASE# {caso}:")

    for _ in range(Q):
        consulta = int(input().strip())

        inicio, fim = 0, len(marmores)
        while inicio < fim:
            meio = (inicio + fim) // 2
            if marmores[meio] < consulta:
                inicio = meio + 1
            else:
                fim = meio

        if inicio >= len(marmores) or marmores[inicio] != consulta:
            print(f"{consulta} not found")
        else:
            print(f"{consulta} found at {inicio + 1}")

File: Clojure/1025 - Onde está o Mármore.clj
(defn binary-search [arr x]
  (loop [inicio 0
         fim (count arr)]
    (if (< inicio fim)
      (let [meio (quot (+ inicio fim) 2)]
        (if (< (nth arr meio) x)
          (recur (inc meio) fim)
          (recur inicio meio)))
      inicio)))

(let [caso (atom 0)]
  (loop []
    (let [entrada (read-line)]
      (when entrada
        (let [[N Q] (map #(Integer/parseInt %) (clojure.string/split entrada #"\s+"))]
          (when-not (and (= N 0) (= Q 0))
            (let [marmores (loop [i 0 acc []]
                             (if (< i N)
                               (recur (inc i) (conj acc (Integer/parseInt (read-line))))
                               (vec (sort acc))))]
              (swap! caso inc)
              (println (format "CASE# %d:" @caso))
              (dotimes [_ Q]
                (let [consulta (Integer/parseInt (read-line))
                      pos (binary-search marmores consulta)]
                  (if (or (>= pos (count marmores))
                          (not= (nth marmores pos) consulta))
                    (println (format "%d not found" consulta))
                    (println (format "%d found at %d" consulta (inc pos)))))))
            (recur)))))))